src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -375,16 +393,16 @@ private function buildToArray(ClassBuilder $class): void
             $code = '$this->PROPERTY';
             if (null !== $p->getType()) {
                 if ($p->isArray()) {
-                    $code = 'array_map(function ($v) { return $v->toArray(); }, $this->PROPERTY)';
+                    $code = 'array_map(function ($v) { return $v instanceof CLASS ? $v->toArray() : $v; }, $this->PROPERTY)';
```

Comment:
can we replace this by `return \is_array($v) ? $v : $v->toArray();` everywhere?

Reply:
@nicolas-grekas Actually no, `$this->PROPERTY` in an array of either `CLASS` objects or scalars, eg:

```php
return static function (ScalarNormalizedTypesConfig $config) {
    $config
        ->listObject('baz')
        ->listObject()->name('qux');
};
```

results in:

```
Symfony\Config\ScalarNormalizedTypesConfig {
  listObject: array:2 [
    0 => "baz"
    1 => Symfony\Config\ScalarNormalizedTypes\ListObjectConfig {
      name: "qux"
      data: null
      _usedProperties: array:1 [
        "name" => true
      ]
    }
  ]
  _usedProperties: array:1 [
    "listObject" => true
  ]
}
```